### ISSUE TITLE
Problems with legend of plot_factors()

### ISSUE THREAD

**Issue by oliviaAB** [bioFAM/MOFA2#83]
Hi,

I am having some issues with the `plot_factors` function. When I try to remove the legend with the `legend` parameter I get the following error message:

```r
> plot_factors(model, factors = 1:3, color_by = "condition", legend = FALSE)
Error in (function (object, factors = "all", show_missing = TRUE, dot_size = 1,  : 
  unused argument (legend = FALSE)
```

Traceback shows that it is due to the `.plot_multiple_factors` which does not have a `legend` argument. 

I am encountering this problem because I am trying to use my own palette for the plot: the plot is displayed with the chosen palette but the legend of the plot still has the original palette (I had to cut the legend labels out of the picture because of sensitive information):

```r
plot_factors(mofa2_output,
             factors = c(3, 5, 9),
             color_by = "other_parent",
             dot_size = 2,
             legend = FALSE) +
  scale_color_brewer(palette = "Paired") +
  scale_fill_brewer(palette = "Paired")
```
![plot_factors_legend](https://user-images.githubusercontent.com/9747181/149243400-5d03c75f-77b9-4d30-ae2f-8ad5b1bd7ad8.png)

Thanks!

**Comment by gtca** [comment]
Hi Olivia!

This should be fixed by fa59952, please check the latest `master` branch if it fixes the problem in your workflow!

**Comment by oliviaAB** [comment]
That works nicely now, thank you for the quick response!
